libraries/Microsoft.Bot.Builder/BotAdapter.cs

```diff
@@ -242,6 +242,33 @@ public virtual Task ContinueConversationAsync(ClaimsIdentity claimsIdentity, Act
             throw new NotImplementedException();
         }
 
+        /// <summary>
+        /// Creates a conversation on the specified channel.
+        /// </summary>
+        /// <param name="botAppId">TThe application ID of the bot.</param>
+        /// <param name="channelId">The ID for the channel.</param>
+        /// <param name="serviceUrl">The channel's service URL endpoint.</param>
+        /// <param name="conversationParameters">The conversation information to use to
+        /// create the conversation.</param>
+        /// <param name="callback">The method to call for the resulting bot turn.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        /// <remarks>To start a conversation, your bot must know its account information
+        /// and the user's account information on that channel.
+        /// Most _channels only support initiating a direct message (non-group) conversation.
+        /// <para>The adapter attempts to create a new conversation on the channel, and
+        /// then sends a <c>conversationUpdate</c> activity through its middleware pipeline
+        /// to the <paramref name="callback"/> method.</para>
+        /// <para>If the conversation is established with the
+        /// specified users, the ID of the activity's <see cref="IActivity.Conversation"/>
+        /// will contain the ID of the new conversation.</para>
+        /// </remarks>
+        public virtual Task CreateConversationAsync(string botAppId, string channelId, string serviceUrl, ConversationParameters conversationParameters, BotCallbackHandler callback, CancellationToken cancellationToken)
```

Comment:
Is there a specific reason we didn't have this method before? Or any of the CreateConversationAsync implementations in `BotFrameworkAdapter`?

Will this support Skills making 1:1 conversations on behalf of the host bot (e.g. in Teams)?

@johnataylor @EricDahlvang

Reply:
@stevengum I'm not sure about Skills, but this should enable CloudAdapter in this Teams scenario: https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/57.teams-conversation-bot/Bots/TeamsConversationBot.cs#L150

Reply:
I think it was just an omission. This improves the signature slightly. Its probably fair to say we simply don't have enough examples to make strong claims around how general any of this is. It improves the typing (removes the casts) by adding it to the abstract class and makes things more consistent. This works for Teams.